omnigan/utils.py

```diff
@@ -29,32 +49,38 @@ def load_opts(path=None, default=None):
     """
     assert default or path
 
+    if path:
+        path = Path(path).resolve()
+
     if default is None:
-        default_opts = Dict()
+        default_opts = {}
     else:
-        with open(default, "r") as f:
-            default_opts = Dict(yaml.safe_load(f))
+        if isinstance(default, (str, Path)):
+            with open(default, "r") as f:
+                default_opts = yaml.safe_load(f)
+        else:
+            default_opts = dict(default)
 
     if path is None:
-        overriding_opts = Dict()
+        overriding_opts = {}
     else:
         with open(path, "r") as f:
-            overriding_opts = Dict(yaml.safe_load(f))
+            overriding_opts = yaml.safe_load(f)
 
-    default_opts.update(overriding_opts)
+    opts = Dict(merge(overriding_opts, default_opts))
```

Comment:
Just wondering why you created a merge() function instead of using dict.update()? :)

Reply:
because for some reason it wasn't recursive so:

```
{a: {b:c, d:e}}.update({a: {b:c, f:g}})
```

would give `{a: {b:c, f:g}` instead of `{a: {b:c, d:e, f:g}}`

Reply:
or something like that :p

Reply:
Oh I see, thanks!